executor/union_scan.go

```diff
@@ -87,6 +91,15 @@ func (us *UnionScanExec) open(ctx context.Context) error {
 		return err
 	}
 
+	for i := len(us.columns) - 1; i >= 0; i-- {
+		if us.columns[i].ID == model.ExtraPhysTblID {
```

Comment:
You mean relying on HandlerID, ExtraPIDColumn and ExtraPhysTblColID is always last?
Sure, but since this is in the `Open` function I think the overhead can be neglected, so I rather have it more generic, in case there will be changes that breaks the assumption of order of the requested columns.